app/services/fetch_link_card_service.rb

```diff
@@ -4,29 +4,40 @@ class FetchLinkCardService < BaseService
   URL_PATTERN = %r{https?://\S+}
 
   def call(status)
-    # Get first http/https URL that isn't local
-    url = parse_urls(status)
+    @status = status
+    @url    = parse_urls
 
-    return if url.nil?
+    return if @url.nil? || @status.preview_cards.any?
 
-    url  = url.to_s
-    card = PreviewCard.where(status: status).first_or_initialize(status: status, url: url)
-    res  = Request.new(:head, url).perform
+    @url  = @url.to_s
+    @card = PreviewCard.find_by(url: @url)
 
-    return if res.code != 200 || res.mime_type != 'text/html'
-
-    attempt_opengraph(card, url) unless attempt_oembed(card, url)
+    process_url if @card.nil?
```

Comment:
I didn't mean to keep that in.

But I think I had a change of heart today morning. The `PreviewCard.find_by(url: @url)` would take an increasing amount of time unless an index is created. So I think I am going to add an index, and it may be unique.